proxy/src/source.rs

```diff
@@ -88,14 +94,14 @@ impl From<radicle_registry_client::Project> for Project {
 }
 
 /// Abstraction used to fetch information from the registry.
-pub trait Source {
+pub trait Source<Error = RegistryError> {
```

Comment:
I think something got lost in translation :sweat_smile: What I was saying that you want to be generic over the `Error` instead of committing to `RegistryError`, which you're doing by use `Error = RegistryError` here.

What I'd probably expect is:
```rust
pub trait Source {
  type Error;
  /* rest of the code */
}
```

Then for here https://github.com/radicle-dev/radicle-upstream/blob/13388f4d3d4f569ab7f981c968abca6d13036d41/proxy/src/source.rs#L157 you would have:
```rust
impl<R> Source for Ledger<R> {
  type Error = RegistryError;
}
```

Reply:
oh, fascinating. that makes more sense.

Reply:
Hence why I mentioned that a "pure" implementation might do `Error = ()` because you're guaranteed to get the data since you control it, for example :)

Reply:
And that would be for testing of course